disdrodb/utils/zip.py

```diff
@@ -13,3 +18,91 @@ def _unzip_file(file_path: str, dest_path: str) -> None:
 
     with zipfile.ZipFile(file_path, "r") as zip_ref:
         zip_ref.extractall(dest_path)
+
+
+def _zip_dir(dir_path: str, files_compression: Optional[str] = None) -> str:
+    """Zip a directory into a file located in the same directory.
+
+    Parameters
+    ----------
+    dir_path : str
+        Path of the directory to zip
+    files_compression : str
+        Compression method for individual files inside directory
+
+    Returns
+    -------
+    str
+        Path of the zip archive
+    """
+    output_path = os.path.join(tempfile.gettempdir(), os.path.basename(dir_path) + ".zip")
+
+    with zipfile.ZipFile(output_path, "w", compression=zipfile.ZIP_DEFLATED) as zipf:
+        for root, dirs, files in os.walk(dir_path):

```

Comment:
I guess we had a misunderstanding here. 
DISDRODB should upload a zipped station directory. The choice if 1) each file within the directory is compressed, or 2) each file is not compressed has to be made by the user before executing the script to upload the data.
The `_zip_dir`  function should just zip the station directory (and do nothing about the files within the directory).

I would re-use the code define within this loop to create the function `compress_station_files(disdrodb_dir, data_source,  campaign_name, station_name)` that compresses each file of a station. This function must be executed by the user before executing the script for uploading the data. The reason is that if each file is compressed, it will have a different extension and the glob_pattern in the station reader function must be updated !

Reply:
Maybe you can use glob to retrieve all file paths, select the ones which are files, then compress each file and replace the uncompressed one. 
Maybe could be also worth adding a function that checks if the files are already compressed (and skip the compression step).

Reply:
Potentially, `compress_station_files(disdrodb_dir, data_source,  campaign_name, station_name)` can be parallelized with multithreading or multiprocessing ...

Reply:
I guess I must add another cli command for calling `compress_station_files`. What would be the name of the command? And in which directory/module do you want me to add the function and the script entry point?

Reply:
@sphamba: let's not complicate our life. We can just provide a python function `compress_station_files` to be called by the user within python. Maybe print at the end of the function a message like: `"Compression of station XXX files completed. Remember to adapt the glob_pattern of the reader accordingly!"`
I would add in this `utils/zip.py` file ... (maybe can be renamed `compression.py`)

Reply:
done